kafka-admin/src/main/java/org/bf2/admin/security/TrustedSSOCallerPrincipalFactoryProducer.java

```diff
@@ -0,0 +1,126 @@
+ * of using a trusted certificate for JWKS retrieval.
+ *
+ * @deprecated needed until https://github.com/smallrye/smallrye-jwt/issues/571
+ *             is fixed and released
+ */
+@Deprecated(forRemoval = true)
+@ApplicationScoped
+@Alternative
+@Priority(Interceptor.Priority.APPLICATION + 10)
+public class TrustedSSOCallerPrincipalFactoryProducer {
+
+    private static final Logger LOG = Logger.getLogger(TrustedSSOCallerPrincipalFactoryProducer.class.getName());
+
+    private JWTCallerPrincipalFactory factory;
+
+    public TrustedSSOCallerPrincipalFactoryProducer() {
+        initCallerPrincipalFactory();
+    }
+
+    @Produces
+    public JWTCallerPrincipalFactory getFactory() {
+        return factory;
+    }
+
+    private void initCallerPrincipalFactory() {
+        LOG.info("Loading JWTCallerPrincipalFactory from " + getClass());
+        factory = new TrustedSSOCallerPrincipalFactory();
+        JWTCallerPrincipalFactory.setInstance(factory);
+    }
+
+    private static class TrustedSSOCallerPrincipalFactory extends DefaultJWTCallerPrincipalFactory {
+        private final DefaultJWTTokenParser parser = new TrustedSSOTokenParser();
+
+        @Override
+        public JWTCallerPrincipal parse(final String token, final JWTAuthContextInfo authContextInfo) throws ParseException {
+            JwtContext jwtContext = parser.parse(token, authContextInfo);
+            String type = jwtContext.getJoseObjects().get(0).getHeader("typ");
+            return new DefaultJWTCallerPrincipal(type, jwtContext.getJwtClaims());
+        }
+    }
+
+    private static class TrustedSSOTokenParser extends DefaultJWTTokenParser {
+        private volatile VerificationKeyResolver keyResolver;
+
+        protected VerificationKeyResolver getVerificationKeyResolver(JWTAuthContextInfo authContextInfo)
+                throws UnresolvableKeyException {
+
+            if (keyResolver == null) {
+                synchronized (this) {
+                    if (keyResolver == null) {
+                        keyResolver = new TrustedSSOKeyLocationResolver(authContextInfo);
+                    }
+                }
+            }
+
+            return keyResolver;
+        }
+    }
+
+    private static class TrustedSSOKeyLocationResolver extends KeyLocationResolver {
+        public TrustedSSOKeyLocationResolver(JWTAuthContextInfo authContextInfo) throws UnresolvableKeyException {
+            super(authContextInfo);
+        }
+
+        @Override
+        protected HttpsJwks initializeHttpsJwks(String location) {
+            HttpsJwks theHttpsJwks = new HttpsJwks(location);
+
+            Config config = ConfigProvider.getConfig();
+            config.getOptionalValue(KafkaAdminConfigRetriever.OAUTH_TRUSTED_CERT, String.class)
+                  .ifPresent(cert -> {
+                      Get httpGet = new Get();
+                      httpGet.setTrustedCertificates(loadPEMCertificate(cert));
```

Comment:
Should there be any error handling around `loadPEMCertificate` or any testing to ensure we loaded a valid/appropriate cert?

Reply:
If `loadPEMCertificate` encounters an exception the result is the same as no certificate being loaded at all. If the JWKS endpoint is trusted by the JVM's trust store, the HTTP request will succeed, else it will fail on the untrusted certificate. 

The method being overridden in this PR `initializeHttpsJwks` is invoked from the parent class's constructor immediately before its return value is used to make the HTTP request. Adding a test for this would require additional overriding of third-party classes that may not be worth it. I don't think it's worth the added complexity. Thoughts?

Reply:
@SamBarker I added a few unit tests and some minor refactoring for the various cert loading scenarios. Please take another look.